adobe/algorithm/selection_algorithms.hpp

```diff
@@ -776,6 +776,208 @@ selection_stable_partition_about(const Selection& selection, ForwardRange& range
     return std::pair<range_iterator, range_iterator>(i, j);
 }
 
+  
+  /****************************************************************************************************/
+  /*!
+   \ingroup selection_algorithms
+   
+   Utility function to split the range into iChunkCount parts
+   
+   Assumes iChunkCount > 0
+   Assumes iterator range distance is > 0
+   
+   
+   Extra check if index set is continuous, we do not perform the union then.
+   */
+  template <typename Iterator>
+  std::vector<std::pair<Iterator, Iterator>>
+  split_range_into_chunks( Iterator iBegin, Iterator iEnd, std::size_t iChunkCount )
```

Comment:
Is it OK to name it ForwardConstIterator? in all of the code, the source index set is not touched.